File: tests/python/solution_files/sol22.py
from __future__ import annotations 
from typing import Optional

class Solution:
  def solve(self, input):
    total = 0
    curr = input
    while curr:
      for a in curr.val:
        for b in a:
          for c in b:
            for d in c:
              total += self.inorder_traverse(d)

      curr = curr.next

    return total
  
  def inorder_traverse(self, root: Optional[TreeNode]) -> int:
    if not root:
      return 0
    ret = 0
    ret += self.inorder_traverse(root.left)
    for a in root.val:
      for b in a:
        if b == True:
          ret += 1
    ret += self.inorder_traverse(root.right)
    return ret
  
class ListNode:
  def __init__(self, val, next: Optional[ListNode] = None):
    self.val = val
    self.next = next 

class TreeNode:
  def __init__(self, val, left: Optional[TreeNode] = None, 
               right: Optional[TreeNode] = None):
    self.val = val
    self.left = left
    self.right = right

File: app_tests/java/solution_files/Sol22.java
package app_tests.java.solution_files;
import java.util.List;
import java.util.Set;

public class Sol22 {
  public static long solve(ListNode input) {
    long total = 0;
    for (TreeNode[][][] a : input.val) {
      for (TreeNode[][] b : a) {
        for (TreeNode[] c : b) {
          for (TreeNode d : c) {
            total += inorderTraverse(d);
          }
        }
      }
    }
    return total;
  }

  private static long inorderTraverse(TreeNode root) {
    if (root == null) {
      return 0;
    }
    long ret = 0;
    ret += inorderTraverse(root.left);
    for (Set<Boolean> a : root.val) {
      for (Boolean b : a) {
        if (b == true) {
          ret += 1;
        }
      }
    }
    ret += inorderTraverse(root.right);
    return ret;
  }

  public static class ListNode {
    public List<TreeNode[][][]> val;
    public ListNode next;

    public ListNode(List<TreeNode[][][]> val, ListNode next) {
      this.val = val;
      this.next = next;
    }
  }

  public static class TreeNode {
    public Set<Boolean>[] val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode(Set<Boolean>[] val, TreeNode left, TreeNode right) {
      this.val = val;
      this.left = left;
      this.right = right;
    }
  }
}
 